packages/pluggableWidgets/color-picker-web/src/components/ColorPicker.tsx

```diff
@@ -0,0 +1,152 @@
+        const alertMessage = validFormat ? invalidFormatMessage?.replace(/\{1}/, validFormat) : undefined;
+        setAlertMessage(alertMessage);
+    };
+    const onToggle = useCallback(
+        (hide: boolean): void => {
+            if (!disabled && mode !== "inline") {
+                setHidden(hide);
+            }
+        },
+        [disabled, mode]
+    );
+    const onColorValueUpdate = useCallback(
+        (color: ColorState): void => {
+            if (color && !disabled) {
+                const finalColor = parseColor(color, format);
+                setCurrentColor(finalColor);
+                props.onChangeComplete(finalColor);
+            }
+        },
+        [disabled, format, props]
+    );
+
+    const onInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        const value = event.target.value;
+        validateColor(color);
+        setCurrentColor(value);
+        props.onChangeComplete(value);
+    };
+
+    const renderInput = (): ReactElement => {
+        return (
+            <Input color={color} disabled={disabled} onKeyUp={onToggle} onChange={onInputChange}>
+                {renderButton()}
+            </Input>
+        );
+    };
+    const renderButton = (): ReactElement => {
+        return <Button mode={mode} hidden={hidden} disabled={disabled} onClick={onToggle} color={color} />;
+    };
+    const renderColorPicker = (): ReactElement => {
+        const supportPopover = mode !== "inline" && type !== "hue" && type !== "slider";
+        return (
+            <div
+                className={classNames({
+                    "widget-color-picker-popover": supportPopover,
+                    "widget-color-picker-no-popover": mode !== "inline"
+                })}
+            >
+                {mode !== "inline" ? (
+                    <div className={"widget-color-picker-cover"} onClick={() => onToggle(true)} />
+                ) : null}
+                {disabled ? <div className={"widget-color-picker-overlay"} /> : null}
+                {createElement(colorElement, { ...config })}
+            </div>
+        );
+    };
+
+    useEffect(() => {
+        const colors = defaultColors.map(color => color.color);
+        const isTriangleAvailable = type === "github" || type === "block" || type === "twitter";
+        const config = {
+            color: currentColor,
+            colors: defaultColors.length > 0 && type !== "swatches" ? colors : undefined,
+            onChangeComplete: props.onChange,
+            presetColors: defaultColors.length > 0 ? colors : undefined,
+            disableAlpha: format !== "rgba",
+            disabled,
+            displayColorPicker: hidden,
+            close: () => onToggle(true),
+            onChange: onColorValueUpdate,
+            ...(isTriangleAvailable && { triangle: "hide" })
+        };
+        setConfig(config);
+        validateColor(color);
+    }, [currentColor, hidden, disabled, format, type, defaultColors, mode]);
```

Comment:
Is it really necessary to have this `useEffect` + put all the config into state instead of passing these as normal props to `colorElement` in L117? As far as I can see, the `config` state is only used for that which I think doesn't have to be in state + `useEffect`.  More than that, it prob causes a lot more re-renders right now cuz of this `useEffect`, which isn't great for performance.

So it would be:
```
createElement(colorElement, {
            color: currentColor,
            colors: defaultColors.length > 0 && type !== "swatches" ? colors : undefined,
            onChangeComplete: props.onChange,
            presetColors: defaultColors.length > 0 ? colors : undefined,
            disableAlpha: format !== "rgba",
            disabled,
            displayColorPicker: hidden,
            close: () => onToggle(true),
            onChange: onColorValueUpdate,
            ...(isTriangleAvailable && { triangle: "hide" })
        });
```

and you can move the `validateColor` to its own `useEffect` that only depends on `color`:
```
useEffect(() => {
  validateColor(color);
}, [color]);
```
which would alos be more readable.